Description: Handle missing athlon architecture gracefully.
Author: Mihai Moldovan <ionic@ionic.de>

On Fedora and Mageia, the default arch list also contains "athlon". If compiled
on other systems, this architecture is omitted. Have to test case handle this
in a smart way.

--- a/tests/hawkey/test_sack.cpp
+++ b/tests/hawkey/test_sack.cpp
@@ -85,8 +85,12 @@ START_TEST(test_list_arches)
     const char ** arches = dnf_sack_list_arches(sack);
 
     /* noarch, x86_64, athlon, i686, i586, i486, i386 */
+    /* athlon missing on non-Fedora or Mageia platforms! */
     fail_unless(g_strv_length((gchar**)arches), 7);
-    ck_assert_str_eq(arches[3], "i686");
+
+    size_t strcmp_ret = (strncmp(arches[3], "i686", strlen("i686")) == 0);
+    strcmp_ret |= (strncmp(arches[3], "i586", strlen("i586")) == 0);
+    ck_assert(strcmp_ret);
 
     g_free(arches);
     g_object_unref(sack);
